rough cut of hpc guide (#505)
CHTC/chtc-website-source: uw-research-computing/hpc-transition-2023.md

---
highlighter: none
layout: markdown-page
title: Transitioning to a New HPC Cluster
published: true
---

The Center for High Throughput Computing’s High Performance Cluster is being 
replaced by a new High Performance System! All components of the system (execute 

> nodes, network, shared file system) are new and we expect excellent performance 
> on the cluster. 

Reviewer: ```suggestion
nodes, network, shared file system) are new and we expect an improved experience for our HPC users.
```

ALL USERS OF THE EXISTING HPC SYSTEM WILL NEED TO MIGRATE TO THIS NEW CLUSTER. 
Importantly, access to the existing cluster will be phased out in early 2022. 
CHTC staff are here to assist with your transition.

## Highlights

* The existing HPC cluster is being replaced by a new cluster. After February 2023 
ALL users will lose access to the existing cluster, and all user files will be 
deleted. 
* Custom software will need to be reinstalled and jobs will need to be tested on 
the new cluster. 
* The `univ2` partition is being renamed, and partition policies have changed. 
* Users should avoid using `mpirun` and instead should use `srun` to execute their 
MPI code.  
* File system conventions have changed - jobs will now use `/scratch/$USER` to run, 
and `/home/$USER` will be mainly used for software installations and reference 
files. 

## Important Dates

* Early January, 2023: New cluster available for general use
* February 14, 2023: Jobs will no longer run on the old cluster
* March 1, 2023: Access to `hpclogin1.chtc.wisc.edu` login node and old file 

> system removed, **Data for all users will be deleted.**

Reviewer: ```suggestion
system removed, **Data for all users will be deleted on the old HPC system.**
```

## What You Need to Do

### Move Data

Back up files from the old cluster to another system (e.g. your laptop), copy 
files you are actively working with to the new cluster, and delete all data off 
the old HPC system. **All files in `/home` and `/software` will be deleted off the 
existing system starting March 1, 2023.**

### Log In and Quota Check

Confirm you can access the new cluster by logging into the new login node. 

### Prepare and Submit Test Jobs

After logging in, prepare and submit a few test jobs to confirm that your work 
will run, paying attention to these important changes: 

1. **Use `/home` and `/scratch` correctly:** jobs should be run out of `/scratch/$USER` 
and the home directory should only be used for software installations and copies 

> of important files and templates. Scratch space has a 100GB quota and home 
> directories have a 20GB quota. The `/software` directory is being phased out. 

Reviewer: ```suggestion
of important files and templates. `\scratch` space has a quota of 100GB disk space and 250,000 items and `\home`
directories have a quota of 20GB disk space and 250,000 items. The `/software` directory is being phased out. 
```

1. **Build software with new modules:** users will need to reinstall and/or rebuild 
their software on the new HPC cluster. Users may encounter different versions of 
common tools on the new cluster, so it is important to try installing your 
software early to ensure compatibility. If a software or library is not available 
that is necessary for your installation is not installed, contact CHTC staff.  

> 1. **Change MPI execution:** Our systems administrators now recommend using `srun` 

Reviewer: ```suggestion
1. **Change MPI execution:** Our systems administrators now recommend using `srun --mpi=pmix` 
```

Reviewer: From Emile: my addition would be to the srun instead of mpirun  section. Direct users to the --mpi=pmix flag with srun.
 srun --mpi=pmix to run MPI programs.

instead of `mpirun` or `mpiexec` to execute MPI type code. 

1. **Change #SBATCH options:** The new cluster has different partition names *and* 
different sized nodes. The main general use partition is now called `shared` 
instead of `univ2` or `univ3`. We also recommend the following changes because 
most of our nodes now have 128 cores, so requesting multiple nodes is not 
advantageous if your jobs are smaller than 128 cores. Here are our recommendations 
for different sized jobs: 

<table>
	<tr>
		<th>Job size</th>
		<th>Recommended <code>#SBATCH</code> flags</th>
	</tr>
	<tr>
		<td>32-128 cores</td>
		<td>Example for 32 cores: <pre>
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=32 # recommend multiples of 16
		</pre></td>
	</tr>
	<tr>
		<td>96 - 256 cores</td>
		<td>Example for 160 cores: <pre>
#SBATCH --nodes=2
#SBATCH --ntasks-per-node=80 # designate cores per node
		</pre> OR:  <pre>
#SBATCH --nodes=2
#SBATCH --ntasks=160 # designate overall cores
		</pre></td>
	</tr>
	<tr>

> 		<td>128 or 256 cores (whole nodes</td>

Reviewer: ```suggestion
		<td>128 or 256 cores (whole nodes)</td>
```

---
highlighter: none
layout: markdown-page
title: Transitioning to a New HPC Cluster
published: true
---

The Center for High Throughput Computing’s High Performance Cluster is being 
replaced by a new High Performance System! All components of the system (execute 
nodes, network, shared file system) are new and we expect an improved experience for our HPC users.

ALL USERS OF THE EXISTING HPC SYSTEM WILL NEED TO MIGRATE TO THIS NEW CLUSTER. 
Importantly, access to the existing cluster will be phased out in early 2022. 

> CHTC staff are here to assist with your transition.

Reviewer 2: It would be nice to highlight this in some way but idk how the HTML gets constructed/deployed. @CannonLock can we do something like mkdocs admonitions (https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types)?

## Highlights

> * The existing HPC cluster is being replaced by a new cluster. After February 2023 

Reviewer 2: ```suggestion
* The existing HPC cluster is being replaced by a new cluster. After **February 2023** 
```

Stress this with bolding
